spacy_llm/tasks/entity_linker/util.py

```diff
@@ -0,0 +1,190 @@
+        """
+        ...
+
+
+@dataclasses.dataclass
+class KBSerializedLoader(BaseInMemoryLookupKBLoader):
+    """Config/init helper class for loading InMemoryLookupKB instance from a serialized KB directory."""
+
+    nlp_path: Optional[Union[str, Path]]
+    """Path to serialized NLP pipeline. If None, path will be guessed."""
+    desc_path: Optional[Union[Path, str]]
+    """Path to .csv file with descriptions for entities. Has to have two columns with the first one being the entity ID,
+        the second one being the description. The entity ID has to match with the entity ID in the stored knowledge
+        base.
+        If not specified, all entity descriptions provided in prompts will be a generic "No description available" or
+        something else to this effect.
+    """
+    ent_desc_reader: EntDescReader
+    """Entity description file reader."""
+
+    def __post_init__(self):
+        super().__post_init__()
+        if self.nlp_path and isinstance(self.nlp_path, str):
+            self.nlp_path = Path(self.nlp_path)
+
+    def __call__(self, vocab: Vocab) -> Tuple[InMemoryLookupKB, DescFormat]:
+        assert isinstance(self.path, Path)
+
+        # Load pipeline, use its vocab. If pipeline path isn't set, try loading the surrounding pipeline
+        # (which might fail).
+        nlp_path = self.nlp_path if self.nlp_path else self.path.parent.parent
+        try:
+            nlp = spacy.load(nlp_path)
+        except IOError as err:
+            raise ValueError(
+                f"Pipeline at path {nlp_path} could not be loaded. Make sure to specify the correct path."
+            ) from err
+        kb = InMemoryLookupKB(nlp.vocab, entity_vector_length=1)
+        kb.from_disk(self.path)
+
+        return kb, self.ent_desc_reader(self.desc_path) if self.desc_path else {}
+
+
+@dataclasses.dataclass
+class KBYamlLoader(BaseInMemoryLookupKBLoader):
+    """Config/init helper class for generating a InMemoryLookupKB instance from a .yaml file."""
+
+    def __call__(self, vocab: Vocab) -> Tuple[InMemoryLookupKB, DescFormat]:
+        assert isinstance(self.path, Path)
+
+        kb_data = srsly.read_yaml(self.path)
+        entities = kb_data["entities"]
+        qids = list(entities.keys())
+        kb = InMemoryLookupKB(
+            vocab=vocab,
+            entity_vector_length=len(
+                kb_data["entities"][qids[0]].get("embedding", [0])
+            ),
+        )
+
+        # Set entities (with dummy values for frequencies).
+        kb.set_entities(
+            entity_list=qids,
+            # Use [0] as default embedding if no embeddings are specified.
+            vector_list=[entities[qid].get("embedding", [0]) for qid in qids],
+            freq_list=[1] * len(qids),
+        )
+
+        # Add aliases and dummy prior probabilities.
+        for alias_data in kb_data["aliases"]:
+            kb.add_alias(**alias_data)
```

Comment:
If the yaml refers to unknown identifiers, spaCy's code will raise an error like 
> ValueError: [E134] Entity 'Q666' is not defined in the Knowledge Base.

While this is pretty helpful, I would consider catching this `ValueError` on this line and add an additional note to the user, saying explicitely that the yaml couldn't be parsed.

I would also consider thinking about similar UX error messages for specific issues in the yaml formatting.

Reply:
Like in https://github.com/explosion/spacy-llm/pull/241/commits/f9991d57f5db7fe5159f641825f391cb348ca309?